src/axom/quest/tests/quest_intersection_shaper.cpp

```diff
@@ -0,0 +1,552 @@
+    double maxdiff = 0.;
+    for(int i = 0; i < a1.number_of_elements() && same; i++)
+    {
+      double diff = fabs(a1[i] - a2[i]);
+      maxdiff = std::max(diff, maxdiff);
+      same &= diff <= tolerance;
+      if(!same)
+      {
+        info.append().set(
+          axom::fmt::format("\"{}\" fields differ at index {}.", n1.name(), i));
+      }
+    }
+    info["maxdiff"][n1.name()] = maxdiff;
+  }
+  else
+  {
+    for(int i = 0; i < n1.number_of_children() && same; i++)
+    {
+      const auto &n1c = n1.child(i);
+      const auto &n2c = n2.fetch_existing(n1c.name());
+      same &= compareConduit(n1c, n2c, tolerance, info);
+    }
+  }
+  return same;
+}
+
+// NOTE: The baselines are read/written using Conduit directly because the
+//       various data collections in Sidre, MFEM, VisIt all exhibited problems
+//       either saving or loading the data.
+void saveBaseline(const std::string &filename, const conduit::Node &n)
+{
+  std::string file_with_ext(filename + ".yaml");
+  SLIC_INFO(axom::fmt::format("Save baseline ", file_with_ext));
+  conduit::relay::io::save(n, file_with_ext, "yaml");
+}
+
+bool loadBaseline(const std::string &filename, conduit::Node &n)
+{
+  bool loaded = false;
+  std::string file_with_ext(filename + ".yaml");
+  SLIC_INFO(axom::fmt::format("Load baseline {}", file_with_ext));
+  // Check before we read because Sidre installs a conduit error handler
+  // that terminates.
+  if(axom::utilities::filesystem::pathExists(file_with_ext))
+  {
+    conduit::relay::io::load(file_with_ext, "yaml", n);
+    loaded = true;
+  }
+  return loaded;
+}
+
+void replacementRuleTest(const std::string &shapeFile,
+                         const std::string &policyName,
+                         int policy,
+                         double tolerance,
+                         bool initialMats = false)
+{
+  // Make potential baseline filenames for this test. Make a policy-specific
+  // baseline that we can check first. If it is not present, the next baseline
+  // is tried.
+  std::string baselineName(yamlRoot(shapeFile));
+  if(initialMats) baselineName += "_initial_mats";
+  std::vector<std::string> baselinePaths;
+  // Example /path/to/axom/src/quest/tests/baseline/quest_intersection_shaper/cuda
+  baselinePaths.push_back(pjoin(baselineDirectory(), policyName));
+  // Example: /path/to/axom/src/quest/tests/baseline/quest_intersection_shaper/
+  baselinePaths.push_back(baselineDirectory());
+
+  // Need to make a target mesh
+  SLIC_INFO(axom::fmt::format("Creating dc {}", baselineName));
+  sidre::MFEMSidreDataCollection dc(baselineName, nullptr, true);
+  makeTestMesh(dc, initialMats);
+
+  // Set up shapes.
+  SLIC_INFO(axom::fmt::format("Reading shape set from {}", shapeFile));
+  klee::ShapeSet shapeSet(klee::readShapeSet(shapeFile));
+
+  // Need to do the pipeline of the shaping driver.
+  SLIC_INFO(axom::fmt::format("Shaping materials..."));
+  const int refinementLevel = 7;
```

Comment:
> const int refinementLevel = 7;

Would it make sense to re-enable the sequential tests, but to decrease the refinement level to something like, 3 or 4? (and possibly the linear segments per knot span)?

Reply:
Also perhaps these parameters can/should be supplied to the test from the command line?

Here's an example of supplying parameters to a test:
https://github.com/LLNL/axom/blob/03d4c74ab518ec397804937b8fc81d2c50a1dd4e/src/axom/quest/examples/CMakeLists.txt#L63-L86

Reply:
Changing the refinement level would probably mean having additional policy-specific baselines. That's something I built in support for but ultimately did not need. Since there's currently an MPI dependency in this program, suppose I divided the tests among ranks? That ought to reduce the time to something manageable.

Reply:
Thanks @BradWhitlock -- sounds good.
This could be done in a follow-up PR.